### DISCUSSION TITLE
EmbedMultipleConfs vs EmbedMolecule

### DISCUSSION THREAD

**Discussion by gcorso** [rdkit/rdkit#5090] [Q&A]
I have a question about the difference between `EmbedMultipleConfs` and calling `EmbedMolecule` multiple times (both part of `rdkit.Chem.rdDistGeom`). Assuming the use of the default parameters of `EmbedMultipleConfs` (therefore, for example, no pruning) is this function simply embedding the molecule independently multiple times or not (i.e. are the distributions of the outputs the same)?
Thank you!

**Comment by greglandrum** [comment]
Yeah, if you aren't doing any pruning then `EmbedMultipleConfs()` is more or less equivalent to calling `EmbedMolecule()` multiple times with `clearConfs` set to false.
It's worth pointing out that `EmbedMultipleConfs()` can be signficantly faster if you allow multiple threads to be used:

compare this:
```
In [30]: params = rdDistGeom.ETKDGv3()

In [31]: params.clearConfs = False

In [32]: e1 = Chem.Mol(ezetimibe)

In [33]: %timeit for i in range(20): rdDistGeom.EmbedMolecule(e1,params)
847 ms ± 19.9 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
```
to this;
```
In [40]: e2 = Chem.Mol(ezetimibe)

In [41]: params.numThreads = 4

In [42]: %timeit rdDistGeom.EmbedMultipleConfs(e2,20,params)
216 ms ± 6.85 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
```

In both cases 20 conformers are generated.
